backend/battle/views.py

```diff
@@ -1,98 +1,71 @@
-#                 round_battle.pk2_creator,
-#                 round_battle.pk3_creator
-#             ]
-#             info = get_total_point_pokemon(data_pokemons)
-#             if info == True:
-#                 round_battle.save()
-#                 return HttpResponseRedirect('/invite/')
-#             else:
-#                 error = message_error()
-#                 return render(request, self.template_name, {'form_round': form_round,
-#                                                              'message': error})
-
-
-# class RoundNewOponnent(View):
-#     form_class = RoundForm2
-#     initial = {'key': 'value'}
-#     template_name = 'battle/round_new2.html'
-#     battle_info = Battle.objects.latest('id')
-
-#     def get(self, request, *args, **kwargs):
-#             form_round2 = self.form_class(initial = self.initial)
-#             battle_get = self.battle_info
-#             return render(request, self.template_name, {'form_round2': form_round2,
-#                                                         'battle': battle_get})
-
-#     def post(self, request, *args, **kwargs):
-#             battle = self.battle_info
-#             form_round2 = self.form_class(request.POST, instance = battle)
-#             if form_round2.is_valid():
-#                 round_opponent = form_round2.save(commit=False)
-#                 data_pokemons = [
-#                     round_opponent.pk1_opponent,
-#                     round_opponent.pk2_opponent,
-#                     round_opponent.pk3_opponent
-#                 ]
-#                 info = get_total_point_pokemon(data_pokemons)
-#                 current_id = battle.id
-#                 if info == True:
-#                     pokemons = [round_opponent.pk1_opponent,
-#                                 round_opponent.pk2_opponent, round_opponent.pk3_opponent]
-#                     result = battleRunning(current_id, pokemons)
-#                     round_opponent.winner = result
-#                     round_opponent.save()
-#                     #result_battle()
-
-#                     return HttpResponseRedirect('/')
-#                 else:
-#                     error = message_error()
-#                     return render(request, self.template_name,
-#                                 {'battle': battle_info, 'message': error})
-
-
-# class Invite(View):
-#     template_name = 'battle/invite.html'
-
-#     def get(self, request):
-#         return render(request, self.template_name)
-
-
-# class Opponent(View):
-#     template_name = 'battle/opponent.html'
-
-#     def get(self, request):
-#         return render(request, self.template_name)
+
+class Invite(TemplateView):
+    template_name = 'battle/invite.html'
+
+
+class BattleView(CreateView):
+    model = Battle
+    template_name = 'battle/battle_form.html'
+    form_class = BattleForm
+
+    def get_initial(self):
+        obj_creator = get_object_or_404(User, email=self.request.user)
```

Comment:
I think this is still wrong, no? `obj_creator = self.request.user` doesn't work?

Reply:
Actually, it works fine, I didn't realize any error with this. Maybe it has something I didn't recognize. Can you explain why this could cause an error?

Reply:
maybe you did get an error when you accessed this page without logging in beforehand.
but either way, it would fail without this as well

Reply:
thank you for the explanation!